packages/react-components/react-search/src/components/SearchBox/useSearchBox.tsx

```diff
@@ -14,14 +20,28 @@ import { DismissRegular, SearchRegular } from '@fluentui/react-icons';
  * @param ref - reference to root HTMLElement of SearchBox
  */
 export const useSearchBox_unstable = (props: SearchBoxProps, ref: React.Ref<HTMLInputElement>): SearchBoxState => {
-  const { size = 'medium', disabled = false, contentBefore, dismiss, contentAfter, ...inputProps } = props;
+  const { size = 'medium', disabled = false, root, contentBefore, dismiss, contentAfter, ...inputProps } = props;
+
+  const searchBoxRootRef = React.useRef<HTMLDivElement>(null);
+  const searchBoxRef = React.useRef<HTMLInputElement>(null);
 
   const [value, setValue] = useControllableState({
     state: props.value,
     defaultState: props.defaultValue,
     initialState: '',
   });
 
+  // Tracks the focus of the component for the contentAfter and dismiss button
+  const [focused, setFocused] = React.useState(false);
+
+  const onFocus = useEventCallback(() => {
+    setFocused(true);
+  });
+
+  const onBlur: React.FocusEventHandler<HTMLSpanElement> = useEventCallback(ev => {
+    setFocused(searchBoxRootRef.current?.contains(ev.relatedTarget) ?? false);
```

Comment:
This works, though you could also do `setFocused(!!searchBoxRootRef.current?.contains(ev.relatedTarget));`